backend/s3afero/single.go

```diff
@@ -158,17 +156,13 @@ func (db *SingleBucketBackend) getBucketWithFilePrefixLocked(bucket string, pref
 func (db *SingleBucketBackend) getBucketWithArbitraryPrefixLocked(bucket string, prefix *gofakes3.Prefix) (*gofakes3.ObjectList, error) {
 	response := gofakes3.NewObjectList()
 
-	if err := afero.Walk(db.fs, filepath.FromSlash(bucket), func(path string, info os.FileInfo, err error) error {
+	if err := afero.Walk(db.fs, filepath.FromSlash("."), func(path string, info os.FileInfo, err error) error {
```

Comment:
Changing from `bucket` to just `"."` seems to drop the bucket path at all, is this correct? If so, could this lead to  unexpected behaviour somewhere? WDYT?

Reply:
If it is `filepath.FromSlash(bucket)`, for example, use `/tmp/fakes3` as s3afero.FsPath, and `mybucket` as bucket name, it will walk through `/tmp/fakes3/mybucket` to scan files. But `PutObject()` actually does not put objects in `/tmp/fakes3/mybucket`, but in `/tmp/fakes3`, so there is conflict, and will cause Internal Server Error when `ListBucket()`.

Reply:
Not sure whether I fully understand it. If `PutObject` is not putting it correctly, shouldn't we change that part? I was not that much involved in the `PutObject` implementation. That's why it could be a stupid question ;-) 

If we cannot or should not change the `PutObject` implementation. It of course makes sense to just start from the "root" path. That's true!

Reply:
I think `PutObject`'s behavior is correct. In the example above, use `/tmp/fakes3` as s3afero.FsPath, and `mybucket` as bucket name, if we put an object `example.txt`, since this is SingleBucket, it is ok to put it at `/tmp/fakes3/example.txt`, rather than `/tmp/fakes3/mybucket/example.txt`.

Reply:
Great! Than good to go!